### PR TITLE
Happy thoughts api 

### PR DESCRIPTION
**Pull request by BamStugendal**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -23,6 +40,37 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// Get thoughts
+app.get('/thoughts', async (req, res) => {
+  const getThoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
```
Comment (rebkova) [inline server.js:45]: To make this part more readable, you could write it like that: 
```
const getThoughts = await Thought.find()
.sort({createdAt: 'desc'})
.limit(20)
.exec()
```

File: server.js [hunk 2]
```diff
@@ -23,6 +40,37 @@ app.get('/', (req, res) => {
...
+  res.json(getThoughts)
```
Comment (rebkova) [inline server.js:46]: Perhaps return an error if the GET request fails?

File: server.js [hunk 3]
```diff
@@ -23,6 +40,37 @@ app.get('/', (req, res) => {
...
+// Post a new thought
+app.post('/thoughts', async (req, res) => {
+  const { message } = req.body
+  const thought = new Thought({ message })
+  
+  try {
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {    
+      console.log(err)
```
Comment (rebkova) [inline server.js:58]: Can be removed. And also in line 69.

File: server.js [hunk 4]
```diff
@@ -23,6 +40,37 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// Get thoughts
+app.get('/thoughts', async (req, res) => {
+  const getThoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(getThoughts)
+})
+
+// Post a new thought
```
Comment (rebkova) [inline server.js:49]: Very nice with some comments 👍

File: server.js [hunk 5]
```diff
@@ -23,6 +40,37 @@ app.get('/', (req, res) => {
...
+// Post like on withthoughtId
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  try {
+    await Thought.updateOne({ _id: req.params.thoughtId}, {$inc: {hearts: 1}})
+    res.status(200).json({ success: true })
+  } catch (err) {
+      console.log(err)
+      res.status(400).json({ success: false , error: err.errors })
+    }
+})
```
Comment (rebkova) [inline server.js:72]: 👍

File: server.js [hunk 6]
```diff
@@ -3,10 +3,27 @@ import bodyParser from 'body-parser'
...
+  createdAt: {
+    type: Date,
+    default: () => new Date()
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  }
+})
+
```
Comment (rebkova) [inline server.js:26]: To the point. 🏹

**Review by rebkova — APPROVED** [review]
Very nice job, Bäm! Pretty straightforward code. I have no major comments.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
You created a very nice project, but I can't test it fully. Please add a link to the frontend to your `README` file.

**Review by KarolCislowski — APPROVED** [review]
Great project 😸

**Pull request closed without merging**